<launch>
  <!-- Arguments turtlebot3_world-->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="slam_methods" default="gmapping" doc="slam type [gmapping, cartographer, hector, karto, frontier_exploration]"/>
  <arg name="configuration_basename" default="turtlebot3_lds_2d.lua"/>
  <arg name="open_rviz" default="true"/>
  <!--include file="$(find turtlebot3_gazebo)/launch/husky.launch">
    <arg name="model" value="$(arg model)" />
  </include-->
  <param name="/error1" type="int" value="0" />
  
  <param name="/s11" type="double" value="0" />
  <param name="/s12" type="double" value="0" />
  <param name="/s13" type="double" value="0" />
  <param name="/s21" type="double" value="0" />
  <param name="/s22" type="double" value="0" />
  <param name="/s23" type="double" value="0" />
  <param name="/a1" type="double" value="0" />
  <param name="/a2" type="double" value="0" />
  
  <!--TurtleBot3-->
  <include file="$(find turtlebot3_gazebo)/launch/turtlebot3_world.launch">
    <arg name="model" value="$(arg model)" />
  </include>

  <node name="map_to_velo_init" pkg="nuise" type="map_helper_ros"/>
  <!--node name="nuise" pkg="nuise" type="nuise"/-->
  
  <!--TurtleBot3>
  <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
    <arg name="model" value="$(arg model)" />
  </include-->

  <!-- SLAM: Gmapping, Cartographer, Hector, Karto, Frontier_exploration, RTAB-Map -->
  <include file="$(find nuise)/launch/turtlebot3_$(arg slam_methods).launch">
    <arg name="model" value="$(arg model)"/>
    <arg name="configuration_basename" value="$(arg configuration_basename)"/>
  </include>
  <!--include file="$(find linefit_ground_segmentation_ros)/launch/segmentation.launch">
  </include-->
  <!--include file="$(find aloam_velodyne)/launch/aloam_velodyne_HDL_32.launch">
  
  </include-->
  
  <!--include file="$(find turtlebot3_teleop)/launch/turtlebot3_teleop_key.launch">
  
  </include-->
  
  <!--node pkg="nuise" type="map_helper_ros" name="map_helper_ros" output="screen">
    <remap from="driver/cmd_vel" to="cmd_vel" />
  </node-->
  <!-- rviz -->
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find turtlebot3_slam)/rviz/turtlebot3_$(arg slam_methods).rviz"/>
  </group>
  <!--include file="$(find aloam_velodyne)/launch/aloam_velodyne_HDL_32.launch">
  </include-->

</launch>
